### PR TITLE
Version bump for cosmos 4.24.0 and spring 3.16.0

### PR DESCRIPTION
**Pull request by simplynaveen20**
Version bump for cosmos 4.24.0 and spring 3.16.0

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,12 @@
 ## Release History
 
-### 4.24.0-beta.1 (Unreleased)
-
+### 4.24.0 (2021-12-21)
 #### Features Added
+* Added implementation for CosmosAuthorizationTokenResolver.
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:5]: CosmosAuthorizationTokenResolver -> `CosmosAuthorizationTokenResolver`

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
+* Scoping session token per partition level for gateway call.
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:6]: scoping -> scoped?

**Review by xinlian12 — APPROVED** [review]
LGTM. Thanks

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 2 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: done

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 1 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: done

**Review by kirankumarkolli — APPROVED** [review]

**Review by milismsft — APPROVED** [review]

**Pull request merged**